### ISSUE TITLE
Black background when plotting brains in PDF

### ISSUE THREAD

**Issue by AlexandreAbraham** [nilearn/nilearn#870]
Steps to reproduce:

* Run this script:
```python
from nilearn.plotting import plot_stat_map
import numpy as np
import nibabel
import pylab as pl


a = nibabel.Nifti1Image(np.zeros((10, 10, 10)), np.eye(4))
plot_stat_map(a)
pl.savefig('test.pdf')
```
* Open `test.pdf` using Inkscape.
* Save it as `test_.pdf`
* Open `test_.pdf` in evince: the background appears black.

**Comment by lesteve** [comment]
Just checking the original test.pdf is fine, right?

**Comment by AlexandreAbraham** [comment]
That depends. If you try to print it, it should be black. I just provided a
way to reproduce it using inkscape not to print one sheet of paper
everytime we want to test a solution.

Alex.

**Comment by GaelVaroquaux** [comment]
> Just checking the original test.pdf is fine, right?

Yes, but if you send it to a printer, it prints like the thing output by
inkscape :(.

Inkscape is just a way of reproducing without relying on a printer.

**Comment by lesteve** [comment]
OK that makes slightly more sense now, thanks.

**Comment by AlexandreAbraham** [comment]
This patch solves the problem: https://github.com/matplotlib/matplotlib/pull/4595/files

It is hardly backportable.

**Comment by GaelVaroquaux** [comment]
> It is hardly backportable.

Darn, matplotlib has compiled code, so we cannot put it in externals.

**Comment by lesteve** [comment]
At least it looks like the fix is in matplotlib 1.5.0: https://github.com/matplotlib/matplotlib/blob/v1.5.0/lib/matplotlib/tests/test_backend_pdf.py#L126.

**Comment by AlexandreAbraham** [comment]
Yes, the bug is solved in 1.5

**Comment by lesteve** [comment]
Maybe we should close this one, since we are not planning to do anything about it.

**Comment by AlexandreAbraham** [comment]
I am actually trying to solve it right now.

**Comment by GaelVaroquaux** [comment]
> I am actually trying to solve it right now.

He is going to solve it. You have no idea!

**Comment by AlexandreAbraham** [comment]
For the record, I tried to backport the PDF backend from matplotlib 1.5.0 into 1.4.3 but there are too many modifications. So I tried to apply the patch mentioned above and, for some reason, it doesn't work.

I'm a bit helpless :-/ on this one.

**Comment by thomasbazeille** [comment]
not sure whether this issue persists. No one complained for a while, will check when i have time

**Comment by jeromedockes** [comment]
tried it now, I can reproduce the issue

**Comment by Remi-Gau** [comment]
Can reproduce
